tools/python/sparsify_initializers.py

```diff
@@ -0,0 +1,134 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# This script opens an existing model in onnx format and attemptes to
+# move initializers from model.graph.initializer field to model.graph.sparse_initializer field
+# and convert them into ONNX COO flat index format.
+
+import argparse
+import logging
+import numpy as np
+import sys
+from typing import Tuple, List
+import onnx
+from onnx import ModelProto, SparseTensorProto, TensorProto, numpy_helper
+
+logger = logging.getLogger(__name__)
+
+real_types = set((np.float32, np.float64, np.double))
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input', required=True, type=str, help='input model path')
+    parser.add_argument('--output', required=True, type=str, help='output model path')
+    parser.add_argument('--exclude', required=False, type=str,
+                        help='semicolon separated list of initializer names to exclude')
+    parser.add_argument('--tolerance', required=False, type=float, default=1e-6,
+                        help='FP tolerance. If not given simple compare to 0')
```

Comment:
Relative or absolute? Or should you have both?

Reply:
Relative to what? We just need to compare to zero.

---
In reply to: [595635191](https://github.com/microsoft/onnxruntime/pull/7004#discussion_r595635191) [](ancestors = 595635191)

Reply:
We don't need relative.